core/discovery/discovery.go

```diff
@@ -44,6 +45,45 @@ const (
 
 const logPrefix = "[discovery] "
 
+// Discovery structure holds discovery service state
+type Discovery struct {
+	identityRegistry            identity_registry.IdentityRegistry
+	ownIdentity                 identity.Identity
+	identityRegistration        identity_registry.RegistrationDataProvider
+	proposalRegistry            ProposalRegistry
+	signerCreate                identity.SignerFactory
+	signer                      identity.Signer
```

Comment:
Do we really need both `identity.Signer` and `identity.SignerFactory`?

Reply:
Existing code, not my scope